cmd/binapi-generator/main.go

```diff
@@ -31,40 +29,53 @@ import (
+		fmt.Fprintf(os.Stderr, "Generating Go bindings for the VPP API\n")
+		fmt.Fprintf(os.Stderr, "--------------------------------------\n\n")
+		fmt.Fprintf(os.Stderr, "This generates VPP api Go bindings based on .api.json files or a VPP repository\n\n")
+		fmt.Fprintf(os.Stderr, "Usage: %s [OPTION]\n", os.Args[0])
 		fmt.Fprintf(os.Stderr, "OPTIONS\n")
 		flag.PrintDefaults()
 		fmt.Fprintf(os.Stderr, "\nEXAMPLES:\n")
-		fmt.Fprintf(os.Stderr, "  %s \\\n", os.Args[0])
-		fmt.Fprintf(os.Stderr, "    --input-dir=$VPP/build-root/install-vpp-native/vpp/share/vpp/api/ \\\n")
-		fmt.Fprintf(os.Stderr, "    --output-dir=~/output \\\n")
-		fmt.Fprintf(os.Stderr, "    interface ip\n")
-		fmt.Fprintf(os.Stderr, "  Assuming --input-dir contains interface.api.json & ip.api.json\n")
+		fmt.Fprintf(os.Stderr, "  Generate bindings from VPP API files (*.api.json):\n")
+		fmt.Fprintf(os.Stderr, "   %s --api /usr/share/vpp/api --output ./myapp/binding --filter interface,ip\n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "  Generate bindings from the VPP repository :\n")
+		fmt.Fprintf(os.Stderr, "   %s --vpp ~/vpp --output ./myapp/binding\n", os.Args[0])
 	}
 }
 
 func printErrorAndExit(msg string) {
-	fmt.Fprintf(os.Stderr, "Error: %s\n\n", msg)
+	color.New(color.FgRed).Fprintf(os.Stderr, "Error:\n%s\n\n", msg)
 	flag.Usage()
 	os.Exit(1)
 }
 
 func main() {
-	var (
-		theApiDir        = flag.String("input-dir", vppapi.DefaultDir, "Input directory containing API files. (e.g. )")
-		theOutputDir     = flag.String("output-dir", "binapi", "Output directory where code will be generated.")
-		importPrefix     = flag.String("import-prefix", "", "Prefix imports in the generated go code. \nE.g. other API Files (e.g. api_file.ba.go) will be imported with :\nimport (\n  api_file \"<import-prefix>/api_file\"\n)")
-		generatorPlugins = flag.String("gen", "rpc", "List of generator plugins to run for files.")
-		theInputFile     = flag.String("input-file", "", "DEPRECATED: Use program arguments to define files to generate.")
-
-		printVersion     = flag.Bool("version", false, "Prints version and exits.")
-		debugLog         = flag.Bool("debug", false, "Enable verbose logging.")
-		noVersionInfo    = flag.Bool("no-version-info", false, "Disable version info in generated files.")
-		noSourcePathInfo = flag.Bool("no-source-path-info", false, "Disable source path info in generated files.")
-	)
+	apiSrcDir := flag.String("input-dir", vppapi.DefaultDir, "[DEPRECATED, use --api] Input directory containing API files.")
+	flag.StringVar(apiSrcDir, "api", "", "Generate based on .api files in this directory.")
+	vppSrcDir := flag.String("vpp", "", "Generate based on a vpp cloned in this directory.")
+
+	// Filtering API files
+	filterList := flag.String("input-file", "", "[DEPRECATED: Use --filter] defines apis to generate.")
+	flag.StringVar(filterList, "filter", "", "Comma separated list of api to generate (e.g. ipip,ipsec, ...)")
```

Comment:
This will require users to write extra flag, making the generator line longer. The arguments to the generator supported using any separator including comma, right? Is there any other reason to add this flag that I am missing?